src/configMap.ts

```diff
@@ -26,10 +26,36 @@ export function loadConfigMapData(obj: KubernetesFileObject) {
     const uri_str = `${uriScheme}://${obj.resource}/${encoded_data}/${obj.id}`;
     const uri = vscode.Uri.parse(uri_str);
     vscode.workspace.openTextDocument(uri).then(
-    (doc) => {
-        vscode.window.showTextDocument(doc);
-    },
-    (error) => {
-        vscode.window.showErrorMessage('Error loading data file: ' + error);
+        (doc) => {
+            vscode.window.showTextDocument(doc);
+        },
+        (error) => {
+            vscode.window.showErrorMessage('Error loading data file: ' + error);
+        });
+}
+
+export async function deleteKubernetesConfigFile(kubectl: Kubectl, obj: KubernetesFileObject, explorer: KubernetesExplorer) {
+    const confirm = 'Delete';
+    let result = await vscode.window.showWarningMessage(`Are you sure you want to delete ${obj.id} this can not be undone`, confirm, 'Cancel');
+    console.log(confirm);
+    if (result !== confirm) {
+        return;
+    }
+    const currentNS = await currentNamespace(kubectl);
+    const json = await kubectl.invokeAsync(`get ${obj.resource} ${obj.parentName} --namespace=${currentNS} -o json`);
+    const dataHolder = JSON.parse(json.stdout);
+    let newData = {};
```

Comment:
I'd like to see lines 47-52 pulled out into a well-named helper so the main function could just read e.g. `dataHolder.data = removeKey(obj.id, dataHolder.data);`

Reply:
done.